### PR TITLE
Add "publish setup" command

### LINKED ISSUE #12: Add publishing to Maven repositories capabilities
coursier has nascent publishing capabilities. These could be used for that, but these need some clean-up / updates before. In particular, it doesn't support "bundle uploads" to Sonatype repositories, that makes publishing there more reliable and that sbt-sonatype uses by default.

### PR DESCRIPTION
**Pull request by alexarchambault**
Fixes https://github.com/VirtusLab/scala-cli/issues/12.

### CODE REVIEW

File: website/docs/commands/publishing/publish-setup.md
Comment (romanowski) [inline website/docs/commands/publishing/publish-setup.md:30]: Shouldn't we also set up github actions with release flow? (snapshot, releases) ?

Comment (alexarchambault) [reply website/docs/commands/publishing/publish-setup.md:30]: You mean, have `scala-cli publish setup` write a workflow file? It does that too, yeah, via the file pulled for https://github.com/VirtusLab/scala-cli/pull/1027.

File: modules/cli/src/main/scala/scala/cli/commands/publish/PublishSetup.scala
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/commands/publish/PublishSetup.scala]: ?

File: modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala:44]: `SharedOptionsOps` already contains an `inputs` method, as well as a few `inputsOrExit`.
Doesn't have to be in the scope of this PR, but I think some scaladocs and/or a refactor here could prevent the future confusion on which way of constructing an `Inputs` instance is the proper one in what circumstances.

File: modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala:42]: the contents of `Foo.scala` could just be `"object Foo"`, no triple-quote and stripping margins, no?

**Review by Gedochao — APPROVED** [review]
Added some NIT comments, none of which should block this PR;
LGTM, although I'd double check if everything works okay after the necessary rebase

File: modules/cli/src/main/scala/scala/cli/commands/publish/PublishSetup.scala
Comment (alexarchambault) [reply modules/cli/src/main/scala/scala/cli/commands/publish/PublishSetup.scala]: That's to print an empty line, for readability. I just pushed some comments for that.

File: modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala
Comment (alexarchambault) [reply modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala:42]: I used that to get an empty line at the end of the file, just in case.

Comment (Gedochao) [reply modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala:42]: ah, okay then.
in that case I'd prefer just:
```scala
"object Foo\n"
```
The alternative is slightly verbose, but I guess it comes down to preference.

Comment (alexarchambault) [reply modules/integration/src/test/scala/scala/cli/integration/PublishSetupTests.scala:42]: Yeah, but on Windows that should be `"object Foo\r\n"`, so at the end, we'd get
```scala
"object Foo" + System.lineSeparator()
```
IMO, multiline + strip margin looks slightly better.

File: modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala
Comment (alexarchambault) [reply modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala:44]: Good point. Just pushed a few commits cleaning that up, and adding a scaladoc to the new `inputs` method.

**Review by romanowski — APPROVED** [review]

**Pull request merged**
